purchase_work_acceptance/models/account_invoice.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    is_required_wa_id = fields.Boolean(
+        default=lambda self: self._get_is_required_wa_id(),
+    )
+    wa_id = fields.Many2one(
+        comodel_name='work.acceptance',
+        string='WA Reference',
+        domain=lambda self: [
+            ('state', '=', 'accept'),
+            ('purchase_id', '=', self._context.get('active_id'))],
+        help='To control quantity and unit price of the vendor bill, to be '
+             'according to the quantity and unit price of the work acceptance.',
+    )
+
+    def _get_is_required_wa_id(self):
+        return self.env.user.has_group(
+            'purchase_work_acceptance.group_work_acceptance_enforce')
+
+    @api.multi
+    def action_invoice_open(self):
+        if self.wa_id:
+            WALines = self.env['work.acceptance.line']
+            for line in self.invoice_line_ids:
+                po_line_id = line.purchase_line_id.id
+                WALine = WALines.search([
```

Comment:
Use lower case for variable, and with 's' to signify multi
```suggestion
                walines = WALines.search([
```

Reply:
Are there other way to work? I think search in every loop is not OK.
Note: When write the code, always imagine that the possibility of having 1,000 lines, and it can be very slow. This way helps you write code better.

Reply:
Again, you can always keep WA data in dict, tuple, list and use it to compare.

Reply:
Don't forget to make it work for multi UoM too.

Reply:
Done